lib/modules/manager/poetry/schema.ts

```diff
@@ -353,19 +354,34 @@ export const PoetryPyProject = Toml.pipe(
 
       deps.push(...dependencyGroups);
 
-      const poetryDependencies = tool?.poetry?.dependencies;
-      if (poetryDependencies) {
-        deps.push(...poetryDependencies);
+      const projectDepsByName: Record<string, PackageDependency> = {};
+      for (const dep of [
+        ...(projectDependencies ?? []),
+        ...(dependencyGroups ?? []),
+        ...(projectOptionalDependencies ?? []),
+      ]) {
+        projectDepsByName[dep.depName!] = dep;
       }
 
+      const poetryDependencies = tool?.poetry?.dependencies;
```

Comment:
```suggestion
      const poetryDependencies = tool?.poetry?.dependencies ?? [];
```

same for other places.